### DISCUSSION TITLE
Help with Vue 3 Composition API Reactive Properties

### DISCUSSION THREAD

**Discussion by alm6666** [mintysolutions/vue_snippets#6] [Help] [answered]
Hello  community!

I'm new to Vue 3 and the Composition API. I'm trying to create a reactive property for a user's name, but I'm not sure if I'm doing it correctly. Here's what I have so far:

```javascript
import { ref } from 'vue'

export default {
  setup() {
    const userName = ref('')

    function updateName(newName) {
      userName = newName  // Is this correct?
    }

    return {
      userName,
      updateName
    }
  }
}

**Comment by dgtalForest** [comment, marked as answer]
Hello @adrilSoft ,  
I'd be happy to help you understand and correct your snippet.

The issue in your code is how you're updating the userName ref. Here's a corrected version of your snippet with explanations:
```
import { ref } from 'vue'

export default {
  setup() {
    const userName = ref('')

    function updateName(newName) {
      userName.value = newName  // Correct way to update a ref
    }

    return {
      userName,
      updateName
    }
  }
}
```

**Reply by alm6666** [reply]
It works, thank you!

**Comment by lodistack** [comment]
@dgtalForest  is correct. It is basics in Vue though :)
